apps/www/registry/default/ui/calendar.tsx

```diff
@@ -1,62 +1,312 @@
+                  0,
+                  1
+                )
+              )
+              return
+            }
+            goToMonth(nextMonth)
+            onNextClick?.(nextMonth)
+          }, [goToMonth, nextMonth])
+          return (
+            <nav className={cn("flex items-center", className)} {...props}>
+              <Button
+                variant="outline"
+                className="absolute left-0 h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100"
+                type="button"
+                tabIndex={isPreviousDisabled ? undefined : -1}
+                disabled={isPreviousDisabled}
+                aria-label={
+                  navView === "years"
+                    ? `Go to the previous ${
+                        displayYears.to - displayYears.from + 1
+                      } years`
+                    : labelPrevious(previousMonth)
+                }
+                onClick={handlePreviousClick}
+              >
+                <ChevronLeft className="h-4 w-4" />
+              </Button>
+
+              <Button
+                variant="outline"
+                className="absolute right-0 h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100"
+                type="button"
+                tabIndex={isNextDisabled ? undefined : -1}
+                disabled={isNextDisabled}
+                aria-label={
+                  navView === "years"
+                    ? `Go to the next ${
+                        displayYears.to - displayYears.from + 1
+                      } years`
+                    : labelNext(nextMonth)
+                }
+                onClick={handleNextClick}
+              >
+                <ChevronRight className="h-4 w-4" />
+              </Button>
+            </nav>
+          )
+        },
+        CaptionLabel: ({ children, ...props }) => {
+          if (!showYearSwitcher) return <span {...props}>{children}</span>
+
+          return (
+            <Button
+              className="h-7 w-full truncate text-sm font-medium"
+              variant="ghost"
+              size="sm"
+              onClick={() =>
+                setNavView((prev) => (prev === "days" ? "years" : "days"))
+              }
+            >
+              {navView === "days"
+                ? children
+                : displayYears.from + " - " + displayYears.to}
+            </Button>
+          )
+        },
+        MonthGrid: ({ className, children, ...props }) => {
+          const { goToMonth } = useDayPicker()
+          if (navView === "years") {
+            return (
+              <div
+                className={cn("grid grid-cols-4 gap-y-2", className)}
+                {...props}
+              >
+                {Array.from(
+                  { length: displayYears.to - displayYears.from + 1 },
+                  (_, i) => {
+                    const isBefore =
+                      differenceInCalendarDays(
+                        new Date(displayYears.from + i, 12, 31),
```

Comment:
Please beware of the second argument being `monthIndex` and is zero-based:

```suggestion
                        new Date(displayYears.from + i, 11, 31),
```